nats/aio/client.py

```diff
@@ -831,6 +831,8 @@ def _process_connect_init(self):
 
         # FIXME: Add readline timeout
         info_line = yield from self._io_reader.readline()
+        if info_line == '':
+            raise NatsError("nats: empty response from server when expecting INFO message")
```

Comment:
Thanks for the contribution! Technically this could occur anytime the connection to the server is reset so same error would occur if the client managed to do a partial read (`INF...`), so think maybe we should wrap in a `try/except` block and raise an error that failed to process initial INFO message, wdyt?